### PR TITLE
[css-ui] Add new test to verify caret-color animation from JavaScript

### PR DESCRIPTION
**Pull request by mrego**
Creating a testharness method so this test can be automatized.
Please @frivoal take a look, thanks.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1153)

### CODE REVIEW

File: css-ui-3/caret-color-018.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Basic User Interface Test: caret-color test animation</title>
+<link rel="author" title="Manuel Rego Casasnovas" href="mailto:rego@igalia.com">
+<link rel="help" href="http://www.w3.org/TR/css3-ui/#caret-color">
+<link rel="help" href="https://www.w3.org/TR/web-animations-1/#dom-animatable-animate">
+<link rel="help" href="https://www.w3.org/TR/css3-color/#color0">
+<meta name="flags" content="dom interact animated">
```
Comment (frivoal) [inline css-ui-3/caret-color-018.html]: * `dom` isn't wrong, but it is not needed for testharness tests (also, we've decided to deprecate it in general, as you generally can infer it from the presence of scripts)
* `animated` is not needed, as the final state of this test is not animated (and we've decided to depracte this flag anyway)
* I don't think you need the `interact` flag here, nothing seems to need interaction in this test.

File: css-ui-3/caret-color-018.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    font-weight: bold;
+    width: 10em;
+    padding: 10px;
+    background: black; color: white; /* the color of a thin object like the caret is easier to see on a black background. */
+
+    caret-color: red;
+  }
+</style>
+<body>
+  <textarea id="textarea" autofocus></textarea>
```
Comment (frivoal) [inline css-ui-3/caret-color-018.html]: I don't think `autofocus` is needed here, so I would suggest removing it. Computed value logic and animations should be running independently of the focus state.

File: css-ui-3/caret-color-018.html (hunk 1 above)
Comment (mrego) [inline css-ui-3/caret-color-018.html]: Ok, I should be more careful regarding the flags next time. Thanks!

File: css-ui-3/caret-color-018.html (hunk 2 above)
Comment (mrego) [inline css-ui-3/caret-color-018.html]: Actually we don't need the styles for the `<textarea>` as this is an automated test.
So I've removed them too.

**Comment by mrego** [comment]
Thanks for the review! Uploading new version.

**Pull request merged**